### PR TITLE
Put responses in pn and pm

### PR DESCRIPTION
**Pull request by maqi**
[Review on Reviewable](https://reviewable.io/reviews/maidsafe/safe_vault/353)

### CODE REVIEW

**Comment by maidsafe-highfive** [comment]
r? @Fraser999

(maidsafe_highfive has picked a reviewer for you, use r? to override)

File: src/personas/pmid_manager.rs [hunk 1]
```diff
@@ -82,77 +84,115 @@ impl Account {
...
+        };
+
+        let src = request.dst.clone();
+        let dst = request.src.clone();
+        trace!("As {:?} sending Put failure to {:?} of data {}", src, dst, data.name());
+        let _ = routing_node.send_put_failure(src, dst, request.clone(), vec![], message_id);
+
+        self.accounts
+            .entry(request.dst.name().clone())
+            .or_insert(Account::default())
```
Comment (Fraser999) [inline src/personas/pmid_manager.rs]: Not sure we should be creating an account for this case?  Probably if the account doesn't exist, just finish.

File: src/personas/pmid_node.rs [hunk 2]
```diff
@@ -134,10 +143,23 @@ impl PmidNode {
...
+
+    pub fn notify_managers_of_success(&mut self,
+                                      routing_node: &RoutingNode,
+                                      data_name: &XorName,
+                                      message_id: &MessageId,
+                                      request: &RequestMessage) -> Result<(), InternalError> {
+        let message_hash = sha512::hash(&try!(serialisation::serialise(&request))[..]);
+        let src = request.dst.clone();
+        let dst = request.src.clone();
+        trace!("As {:?} sending put success of data {} to {:?}", src, data_name, dst);
```
Comment (Fraser999) [inline src/personas/pmid_node.rs]: Could the two trace statements for Put success and failure have the same order of args please?

Comment (maqi) [inline src/personas/pmid_node.rs]: done

**Comment by Fraser999** [comment]
Reviewed 2 of 4 files at r1, 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/safe_vault/353)*

**Comment by Fraser999** [comment]
Reviewed 1 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/maidsafe/safe_vault/353)*

**Pull request merged**